Source/WebCore/css/CSSFontSelector.cpp

```diff
@@ -245,6 +246,23 @@ void CSSFontSelector::addFontPaletteValuesRule(StyleRuleFontPaletteValues& fontP
     ++m_version;
 }
 
+void CSSFontSelector::addFontFeatureValuesRule(const StyleRuleFontFeatureValues& fontFeatureValuesRule)
+{
+    Ref<FontFeatureValues> fontFeatureValues = fontFeatureValuesRule.value();
+
+    for (auto fontFamily : fontFeatureValuesRule.fontFamilies()) {
+        auto initial = FontFeatureValues::create();
+        auto exist = m_featureValues.find(fontFamily);
+        if (exist != m_featureValues.end())
+            initial = exist->value;
+
+        initial->updateOrInsert(fontFeatureValues);
+        m_featureValues.set(fontFamily, initial);
```

Comment:
This is the wrong idiom for updating in place, inefficient and kind of wordy too. Instead we should use `add` or `ensure` and then call `updateOrInsert(fontFeatureValues)` on the value in place inside the hash table.